corehq/blobs/pillow.py

```diff
@@ -19,46 +14,28 @@
 
 class BlobDeletionProcessor(PillowProcessor):
 
-    def __init__(self, blob_db, bucket_base):
+    def __init__(self, blob_db):
         super(BlobDeletionProcessor, self).__init__()
         self.blob_db = blob_db
-        self.bucket_base = bucket_base
 
     def process_change(self, pillow_instance, change):
         if change.deleted:
-            bucket = join(self.bucket_base, change.id)
+            bucket = "{}/{}".format(change.meta.data_source_name, change.id)
```

Comment:
and sorry, here's [where the kafka message gets populated from couch](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/change_feed/pillow.py#L33).

though looking at this I think there might be a problem with my suggestion which is that I don't think hard deletions are properly pushed to kafka. I think to fix that we need to check for `change.deleted` and make a metadata object with the right id and data source info so that the other pillows can pick it up. let me know if that makes sense and is something you'd be willing to add to this PR - otherwise I think we can put it on the backlog (or maybe @snopoke will volunteer to give it a shot)